org.eclipse.zest.core/src/org/eclipse/zest/core/viewers/internal/AbstractStructuredGraphViewer.java

```diff
@@ -250,12 +250,23 @@ GraphNode addGraphModelNode(IContainer container, Object element) {
 		}
 		return node;
 	}
+	
+	/** Implement and return the new node object, enables to define custom graph nodes*/
+	protected GraphNode createNodeObject(final Graph graphModel, final int style, IFigure figure) {
```

Comment:
unfortunately with your update the since tag is missing can you please also add some descrpition of the parameters and the return value